tests/tests-hz/src/main/java/com/hazelcast/simulator/tests/icache/IntByteICacheTest.java

```diff
@@ -0,0 +1,148 @@
+import java.util.concurrent.TimeoutException;
+
+import static com.hazelcast.simulator.tests.helpers.KeyUtils.generateIntKeys;
+import static com.hazelcast.simulator.tests.icache.helpers.CacheUtils.getCache;
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateByteArray;
+
+public class IntByteICacheTest extends HazelcastTest {
+
+    // properties
+    public int keyCount = 1000;
+    public int valueCount = 1000;
+    public int minSize = 16;
+    public int maxSize = 2000;
+    // the number of keys that are going to be written.
+    // normally you want to keep this the same as keyCount (for reading), but it can help to expose certain problems like
+    // gc. If they writeKeyCount is very small, only a small group of objects get updated frequently and helps to prevent
+    // getting them tenured. If writeKeyCount is -1, it will automatically be set to keyCount
+    public int writeKeyCount = -1;
+    public KeyLocality keyLocality = KeyLocality.SHARED;
+    // timeout in milliseconds for Cache asynchronous operations
+    public int timeout = 50;
+    // if true, any occurrence of TimeoutException is caught and only logged, if false it's propagated further
+    public boolean catchTimeoutException = false;
+
+    private ICache<Integer, Object> cache;
+    private int[] keys;
+    private byte[][] values;
+
+    @Setup
+    public void setUp() {
+        cache = getCache(targetInstance, name);
+        keys = generateIntKeys(keyCount, keyLocality, targetInstance);
+
+        if (minSize > maxSize) {
+            throw new IllegalStateException("minSize can't be larger than maxSize");
+        }
+
+        if (writeKeyCount == -1) {
+            writeKeyCount = keyCount;
+        }
+    }
+
+    @Prepare
+    public void prepare() {
+        Random random = new Random();
+        values = new byte[valueCount][];
+        for (int i = 0; i < values.length; i++) {
+            int delta = maxSize - minSize;
+            int length = delta == 0 ? minSize : minSize + random.nextInt(delta);
+            values[i] = generateByteArray(random, length);
+        }
+
+        Streamer<Integer, Object> streamer = StreamerFactory.getInstance(cache);
+        for (int key : keys) {
+            streamer.pushEntry(key, values[random.nextInt(values.length)]);
+        }
+        streamer.await();
+    }
+
+    @TimeStep(prob = 0.0)
+    public void put(ThreadState state) {
+        cache.put(state.randomKey(), state.randomValue());
+    }
+
+    @TimeStep(prob = 0.1)
+    public void putAsync(ThreadState state) throws InterruptedException, ExecutionException, TimeoutException {
+        Future<Void> future = cache.putAsync(state.randomKey(), state.randomValue());
+        try {
+            future.get(timeout, TimeUnit.MILLISECONDS);
+        } catch (TimeoutException ex) {
+            if (catchTimeoutException) {
+                logger.error("putAsync didn't finish within configured " + timeout + " ms timeout.", ex);
```

Comment:
Can you add a counter so we can see how often this has happeend? So an AtomicLong field.